crates/fuel-core/src/graphql_api/storage.rs

```diff
@@ -249,4 +249,53 @@ where
         self.commit()?;
         Ok(())
     }
+
+    #[cfg(feature = "fault-proving")]
+    fn registry_root(&self) -> StorageResult<Bytes32> {
+        use fuel_core_storage::tables::merkle::DenseMetadataKey;
+
+        use crate::graphql_api::storage::da_compression::v2::{
+            address::TemporalRegistryAddressMerkleMetadata as AddressMerkle,
+            asset_id::TemporalRegistryAssetIdMerkleMetadata as AssetIdMerkle,
+            contract_id::TemporalRegistryContractIdMerkleMetadata as ContractIdMerkle,
+            evictor_cache::TemporalRegistryEvictorCacheMerkleMetadata as EvictorCacheMerkle,
+            predicate_code::TemporalRegistryPredicateCodeMerkleMetadata as PredicateCodeMerkle,
+            registry_index::TemporalRegistryIndexMerkleMetadata as RegistryIndexMerkle,
+            script_code::TemporalRegistryScriptCodeMerkleMetadata as ScriptCodeMerkle,
+            timestamps::TemporalRegistryTimestampsMerkleMetadata as TimestampsMerkle,
+        };
+
+        macro_rules! get_merkle_root {
+            ($self:ident, $metadata:ty) => {
+                $self
+                    .storage::<$metadata>()
+                    .get(&DenseMetadataKey::Latest)?
```

Comment:
I went down the rabbit hole of trying to understand this `DenseMetadataKey` which seems to serve the purpose of versioning the merkle tree. Right now we use `RegistryKey` which feels wrong, so while unrelated to the current PR I think we should create a follow-up where we change that to be the block height instead.

Reply:
hmm, you're right. i was under the impression that we have to use the same key as that's used in the data table as the generic, and not an arbitrary type.

Reply:
but it's versioned internally under u64 so i don't think it will be an issue

Reply:
Yeah I don't think this will blow up in our faces, and if it does we'll discover it later when hooking things up together.